rust/blockstore/src/types/reader.rs

```diff
@@ -113,11 +113,11 @@ impl<
         }
     }
 
-    pub async fn load_blocks_for_keys(&self, prefixes: &[&str], keys: &[K]) {
+    pub async fn load_blocks_for_keys(&self, keys: impl IntoIterator<Item = (String, K)>) {
```

Comment:
small change to make the API nicer to work with and put the burden of owned data on the caller